packages/ui/src/components/ChatMessage/ChatMessageActions.tsx

```diff
@@ -0,0 +1,279 @@
+        <YStack gap="$xs">
+          <EmojiToolbar
+            post={post}
+            onDismiss={onDismiss}
+            currentUserId={currentUserId}
+          />
+          <MessageContainer post={post} currentUserId={currentUserId} />
+          <ChatMessageActionsList
+            post={post}
+            channelType={channelType}
+            dismiss={onDismiss}
+          />
+        </YStack>
+      </View>
+    </Animated.View>
+  );
+}
+
+export function EmojiToolbar({
+  post,
+  currentUserId,
+  onDismiss,
+}: {
+  post: db.PostWithRelations;
+  currentUserId: string;
+  onDismiss: () => void;
+}) {
+  const [sheetOpen, setSheetOpen] = useState(false);
+  const details = useReactionDetails(post.reactions ?? [], currentUserId);
+
+  const handlePress = useCallback(async (shortCode: string) => {
+    details.self.didReact && details.self.value.includes(shortCode)
+      ? store.removePostReaction(post.channelId, post.id, currentUserId)
+      : store.addPostReaction(
+          post.channelId,
+          post.id,
+          shortCode,
+          currentUserId
+        );
+
+    await Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success);
+    setTimeout(() => onDismiss(), 50);
+  }, []);
+
+  const lastShortCode =
+    details.self.didReact &&
+    !['+1', 'heart', 'cyclone', 'seedling'].some((code) =>
+      details.self.value.includes(code)
+    )
+      ? details.self.value
+      : 'seedling';
+
+  return (
+    <>
+      <XStack
+        padding="$l"
+        backgroundColor="$background"
+        borderRadius="$l"
+        justifyContent="space-between"
+        alignItems="center"
+        width={256}
+      >
+        <EmojiToolbarButton
+          details={details}
+          shortCode="+1"
+          handlePress={handlePress}
+        />
+        <EmojiToolbarButton
+          details={details}
+          shortCode="heart"
+          handlePress={handlePress}
+        />
+        <EmojiToolbarButton
+          details={details}
+          shortCode="cyclone"
+          handlePress={handlePress}
+        />
+        <EmojiToolbarButton
+          details={details}
+          shortCode={lastShortCode}
+          handlePress={handlePress}
+        />
+        <Button
+          padding="$xs"
+          borderWidth={0}
+          onPress={() => setSheetOpen(true)}
+        >
+          <Icon type="ChevronDown" size="$l" />
+        </Button>
+      </XStack>
+      <EmojiPickerSheet
+        open={sheetOpen}
+        onOpenChange={() => setSheetOpen(false)}
+        onEmojiSelect={handlePress}
+      />
+    </>
+  );
+}
+
+function EmojiToolbarButton({
+  shortCode,
+  details,
+  handlePress,
+}: {
+  shortCode: string;
+  details: ReactionDetails;
+  handlePress: (shortCode: string) => void;
+}) {
+  return (
+    <Button
+      padding="$xs"
+      borderWidth={0}
+      backgroundColor={
+        details.self.didReact && details.self.value.includes(shortCode)
```

Comment:
Should use `$positiveBackground` and `$positiveBorder` -- might need a better name for that.

Reply:
I think the name is clear, this was just sloppy.